### PR TITLE
Implementing searching for officers 

### PR DESCRIPTION
**Pull request by kevbite**
Connects #34

### CODE REVIEW

File: src/LiberisLabs.CompaniesHouse/Request/SearchRequest.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿namespace LiberisLabs.CompaniesHouse.Request
 {
-    public class CompanySearchRequest
+    public class SearchRequest
```
Comment (Blundell89) [inline src/LiberisLabs.CompaniesHouse/Request/SearchRequest.cs]: Breaking change GAAAAHHHHHHHHHHHH

File: src/LiberisLabs.CompaniesHouse/ICompaniesHouseCompanySearchClient.cs [hunk 2]
```diff
@@ -5,8 +5,8 @@
 
 namespace LiberisLabs.CompaniesHouse
 {
-    public interface ICompaniesHouseSearchCompanyClient
+    public interface ICompaniesHouseCompanySearchClient
```
Comment (Blundell89) [inline src/LiberisLabs.CompaniesHouse/ICompaniesHouseCompanySearchClient.cs]: No need to rename?

File: src/LiberisLabs.CompaniesHouse/CompaniesHouseClientResponse.cs [hunk 3]
```diff
@@ -1,3 +1,5 @@
+using System.Runtime.InteropServices.ComTypes;
```
Comment (Blundell89) [inline src/LiberisLabs.CompaniesHouse/CompaniesHouseClientResponse.cs]: ???

File: appveyor.yml [hunk 4]
```diff
@@ -3,8 +3,7 @@
     only:
       - master
 
-  version: 1.5.1
-
+  version: 1.6.0
```
Comment (Blundell89) [inline appveyor.yml]: If breaking changes are included --> `2.0.0`

File: src/LiberisLabs.CompaniesHouse/SearchUriBuilderFactory.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using LiberisLabs.CompaniesHouse.Response.CompanySearch;
+using LiberisLabs.CompaniesHouse.Response.OfficerSearch;
+using LiberisLabs.CompaniesHouse.UriBuilders;
+
+namespace LiberisLabs.CompaniesHouse
+{
+    public class SearchUriBuilderFactory : ISearchUriBuilderFactory
+    {
+        public ISearchUriBuilder Create<TSearch>()
```
Comment (Blundell89) [inline src/LiberisLabs.CompaniesHouse/SearchUriBuilderFactory.cs]: Could add a marker interface to `TSearch` implementations

File: src/LiberisLabs.CompaniesHouse/Response/OfficerSearch/Officer.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        public string[] DescriptionIdentifiers { get; set; }
+
+        [JsonProperty(PropertyName = "kind")]
+        public string Kind { get; set; }
+
+        [JsonProperty(PropertyName = "links")]
+        public Links Links { get; set; }
+
+        [JsonProperty(PropertyName = "matches")]
+        public Match Matches { get; set; }
```
Comment (Blundell89) [inline src/LiberisLabs.CompaniesHouse/Response/OfficerSearch/Officer.cs]: :( @ the documentation being wrong https://developer.companieshouse.gov.uk/api/docs/search-overview/OfficerSearch-resource.html

File: src/LiberisLabs.CompaniesHouse/SearchUriBuilderFactory.cs (hunk 5 above)
Comment (kevbite) [inline src/LiberisLabs.CompaniesHouse/SearchUriBuilderFactory.cs]: Shouldn't really be able to change this externally anyway

**Pull request merged**